systemtest/approvals/TestAgentConfig.approved.json

```diff
@@ -1,5 +1,36 @@
 {
     "events": [
+        {
+            "@timestamp": "dynamic",
+            "agent_config_applied": 1,
+            "data_stream": {
+                "dataset": "apm.internal",
+                "namespace": "default",
+                "type": "metrics"
+            },
+            "ecs": {
+                "version": "dynamic"
+            },
+            "event": {
+                "agent_id_status": "missing",
+                "ingested": "dynamic"
+            },
+            "labels": {
+                "etag": "dynamic"
+            },
+            "metricset": {
+                "name": "agent_config"
+            },
+            "observer": {
+                "hostname": "dynamic",
+                "type": "apm-server",
+                "version": "dynamic"
+            },
+            "processor": {
+                "event": "metric",
+                "name": "metric"
+            }
+        },
```

Comment:
Why are there two metricsets now? The related [comment](https://github.com/elastic/apm-server/blob/main/systemtest/agentconfig_test.go#L118-L120) explicitly mentions only one metricset.

Reply:
The comment was updated to "one metricset per server" in the PR. Since we are starting 3 servers here, 1 elasticsearch-based agentcfg apm server, 1 kibana-based agentcfg apm server, and 1 apm-integration agentcfg apm server, it will result in 1 metric set for elasticsearch, 1 for kibana, and 0 for apm-integration due to the known bug that the metricset cannot be flushed due to apm server restart on config change (which is noted in the comment). Therefore we are expecting 2 metricsets.

Reply:
Makes sense! I missed that.